lib/fog/azurerm/docs/storage.md

```diff
@@ -176,42 +166,119 @@ result = container.destroy
 puts "#{result}"
 ```
 
-### Metadata
+## Upload a local file as a blob
+```ruby
+new_blob = azure_storage_service.upload_block_blob_from_file('<container name>', '<blob name>', '<file path>')
+puts "#{new_blob.inspect}"
+```
+
+## Download a blob to a local file
+```ruby
+blob = azure_storage_service.download_blob_to_file('<container name>', '<blob name>', '<file path>')
+puts "#{blob.inspect}"
+puts "File Size: #{File.size <file_path>}"
+```
+
+## Delete the storage blob
+
+Mark the specified blob for deletion. The blob is later deleted during garbage collection.
+
+```ruby
+blob = azure_storage_service.blobs(container_name: '<container name>').get('<blob name>')
+result = blob.destroy
+puts "#{result}"
+```
+
+Note that in order to delete a blob, you must delete all of its snapshots.
+
+```ruby
+blob = azure_storage_service.blobs(container_name: '<container name>').get('<blob name>')
+result = blob.destroy(delete_snapshots: 'only')
+puts "#{result}"
+
+result = blob.destroy
+puts "#{result}"
+```
+
+You can delete both at the same time by specifying the option.
+
+```ruby
+blob = azure_storage_service.blobs(container_name: '<container name>').get('<blob name>')
+result = blob.destroy(delete_snapshots: 'inlcude')
+puts "#{result}"
+```
+
+## Properties
+
+### Get storage container properties
+
+Get the storage container properties. The properties will not fetch the access control list. Call `get_container_access_control_list` to fetch it.
+
+```ruby
+container = azure_storage_service.containers.get('<container name>')
+properties = container.get_properties
+puts "#{properties.inspect}"
+```
+
+### Get storage blob properties
+
+Get the storage blob properties.
+
+```ruby
+blob = azure_storage_service.blobs(container_name: '<container name>').get('<blob name>')
+properties = blob.get_properties
```

Comment:
This request updates/full-fills the properties available for the blob. 
`azure_storage_service.blobs.get(container_name, blob_name)` doesn't have all the properties filled.